File: app/src/main/java/com/waveneuro/views/WaveProgressDialog.java
package com.waveneuro.views;

import android.app.Activity;
import android.app.AlertDialog;
import android.graphics.drawable.ColorDrawable;
import android.view.LayoutInflater;
import android.view.View;

import com.waveneuro.R;

public class WaveProgressDialog {
    private Activity activity;
    private AlertDialog alertDialog;

    public WaveProgressDialog(Activity activity) {
        this.activity = activity;
    }

    public void startProgressDialog() {
        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(activity);
        LayoutInflater inflater = activity.getLayoutInflater();
        View dialogView = inflater.inflate(R.layout.custom_progress, null);
        dialogBuilder.setView(dialogView);
        dialogBuilder.setCancelable(false);

        alertDialog = dialogBuilder.create();
        alertDialog.getWindow().setBackgroundDrawable(new ColorDrawable(android.graphics.Color.TRANSPARENT));
        alertDialog.show();
    }

    public void dismiss() {
        alertDialog.dismiss();
    }

    public boolean isShowing() {
        if(alertDialog == null)
            return false;
        return alertDialog.isShowing();
    }
}

File: app/src/main/java/com/waveneuro/views/WaveProgressDialog.kt
package com.waveneuro.views

import android.app.Activity
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import androidx.appcompat.app.AlertDialog
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.waveneuro.databinding.CustomProgressBinding
import com.waveneuro.databinding.SessionProgressBinding

fun Activity.waveProgressDialogBuilder() : AlertDialog {
    val binding = CustomProgressBinding.inflate(layoutInflater)
    val dialogBuilder = AlertDialog.Builder(this).setView(binding.root)
    dialogBuilder.setCancelable(false)
    val dialog = dialogBuilder.create()
    dialog.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))

    return dialog
}

fun Activity.sessionProgressDialogBuilder() : AlertDialog {
    val binding = SessionProgressBinding.inflate(layoutInflater)
    val dialogBuilder = MaterialAlertDialogBuilder(this).setView(binding.root)
    dialogBuilder.setCancelable(false)
    val dialog = dialogBuilder.create()
    dialog.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
    dialog.window?.setDimAmount(0.8f)

    return dialog
}